Tella/Domain/Entity/CommonServer/Server.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  Server.swift
+//  Tella
+//
+//  Created by Dhekra Rouatbi on 28/6/2024.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+
+protocol ServerProtocol: Codable, Equatable, Hashable {
```

Comment:
maybe we don't need a `ServerProtocol` in this case, as the parent class `Server` already includes all the parameters that `ServerProtocol` contains.
What do you think? 
cc @Rimktarii